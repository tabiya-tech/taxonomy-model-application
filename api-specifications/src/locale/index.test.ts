describe("Test the locale module", () => {
  test("The locale module can be required via the index", async () => {
    // GIVEN the module
    // WHEN the module is required via the index
    expect(async () => await import("./")).not.toThrow(); // We check that it doesn't throw an error instead of simply letting it fail on import because we want an easier error message
    const localeModule = await import("./");

    // THEN check if Schema is defined in it
    expect(localeModule.default.Schemas.Payload).toBeDefined();
    // AND check if constants are defined in it
    expect(localeModule.default.Constants.LOCALE_SHORTCODE_MAX_LENGTH).toBeDefined();
    expect(localeModule.default.Constants.NAME_MAX_LENGTH).toBeDefined();
  });

  test("The locale module matches the snapshot", () => {
    // GIVEN the module
    // WHEN the module is required via the index
    expect(async () => {
      // THEN Check if the module can be required without error
      const localeModule = await import("./");
      expect(localeModule.default).toMatchSnapshot();
    }).not.toThrowError();
  });
});

export {};
