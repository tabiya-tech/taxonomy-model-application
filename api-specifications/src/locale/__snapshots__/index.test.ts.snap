// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the locale module The locale module matches the snapshot 1`] = `
{
  "Constants": {
    "LOCALE_SHORTCODE_MAX_LENGTH": 20,
    "NAME_MAX_LENGTH": 256,
  },
  "Schemas": {
    "Payload": {
      "$id": "/components/schemas/LocaleSchema",
      "additionalProperties": false,
      "properties": {
        "UUID": {
          "description": "The UUID of the locale",
          "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$",
          "type": "string",
        },
        "name": {
          "description": "The name of the locale",
          "maxLength": 256,
          "pattern": "\\S",
          "type": "string",
        },
        "shortCode": {
          "description": "The short code of the locale",
          "maxLength": 20,
          "pattern": "\\S",
          "type": "string",
        },
      },
      "required": [
        "UUID",
        "shortCode",
        "name",
      ],
      "type": "object",
    },
  },
  "Types": {},
}
`;
